### ISSUE TITLE
Is it possible that in anyway that numbers could be a number in the model value

### ISSUE THREAD

**Issue by maccurt** [uhlryk/angular-dynamic-number#30]
my ngmodel value is currently returning as a string. When I was using the input type of number it would always return the value as a number. Obviously you probably can't use the type of number, but could it somehow parse the value as float. I am using my values in math functions and sometimes the string values don't work. I have a work around, but just wondered if that was possible.

**Comment by uhlryk** [comment]
Yes this is possible and very easy.
But there is problem with compatibility.  By default [input type="text"] for directives returns strings. Other users might use other directives together with this one (this is not supported but it is possible), or do some operations on this strings.
Therefore it is very possible that changing string to number will fail many projects.
Moreover changing strings to numbers is not to complicated in in most cases it is done by automatically by JavaScript itself.

In summary I can do it without problem but Im afraid. If anybody has cons and pros of both solutions please write them below. I will consider them and if we change this, I will do it in next major/minor version.

**Comment by maccurt** [comment]
Perhaps you could leave it working as it is by default and add another option such as parse-as-float, or parse-number.. that way it works for previous users, but other people can change the behavior. From what I see this directive is mainly for accepting numbers in a text box, so I was hoping it would return a number. I highly appreciate the way it is working now, If you can't do that I can change my function to parse it as float, OR I will tinker with your code  to make it parse float. I don't want to do that because I would like anyone to pull my code and have the right version of your directive installed from bower, etc.. By the way this is an AWESOME directive.

**Comment by uhlryk** [comment]
> another option such as parse-as-float or parse-number..

good idea and this does't add breaking changes. I will do it at weekend.

Im very happy that you like this directive. And thank you very much for your feedback.

**Comment by func0der** [comment]
I am just gonna get into this issue with my concern about output normalization, because this is very related.

The normalization you did here: https://github.com/uhlryk/angular-dynamic-number/commit/eea8ac64c67afdd6efe6dd4e149dcf133e226716
 holds the tiny problem that you can not check for `!modelName` in other directives anymore, because `!!'0' === true`. Before it was `!!0 === false`.
This breaks, what I consider as, 'normal model behavior'.

Also, considering that we are handling numbers here, the model should have numeric values at all time after the directive has been triggered. The input holding the value still needs strings, of course.

By the way: This is in my opinion not a change for a patch version change, but at least a minor version change, because of JS handling those two value completely different. Maybe you should revert it and roll it with a dedicated new minor version.

**Comment by maccurt** [comment]
I do believe also it should return a number by default, but I was willing to compromise considering this directive saved me countless hours.. Angular 2 anybody?

**Comment by uhlryk** [comment]
Ok, I will today revert commit mentioned by @func0der and create major version which pass only numbers to models.

@maccurt about Angular2. I plan to isolate logic of this directive as separate repository and then use this project as wrapper, and create angular2 wrapper too.

**Comment by uhlryk** [comment]
v1.12.5 has reverted changes from https://github.com/uhlryk/angular-dynamic-number/commit/eea8ac64c67afdd6efe6dd4e149dcf133e226716

v2.0.0 has for model type number

Please check if this is what you need.

**Comment by maccurt** [comment]
I will try it this morning, THANK YOU VERY MUCH..This is for my personal website, but we have the same need at my Job, and plenty of teams have already told me they plan to use it (the 1.x angular guys).. I plan to spread it's use far and wide.. I really think it should just be built into HTML input by default.. Michael C. McGee (AKA Barnabas)Encouragement  & Strategy DepartmentProfitDreamer.comBigMuddyFishing.com

```
  From: Krzysztof Sztompka <notifications@github.com>
```

 To: uhlryk/angular-dynamic-number angular-dynamic-number@noreply.github.com 
Cc: maccurt dineromike@yahoo.com; Mention mention@noreply.github.com
 Sent: Saturday, August 13, 2016 3:04 PM
 Subject: Re: [uhlryk/angular-dynamic-number] Is it possible that in anyway that numbers could be a number in the model value (#30)

v1.12.5 has reverted changes from eea8ac6v2.0.0 has for model type numberPlease check if this is what you need.—

**Comment by maccurt** [comment]
I just wanted to let you know that is worked for me. I was able to remove all my code in my services that was parsing the input into a float. Thanks..

**Comment by uhlryk** [comment]
Great! Im very happy that this directive is useful for you! And thank you for your kind words.

**Comment by maccurt** [comment]
Can you give me an example of how to use:
num-round:Define round method for fraction part when convert from model to view and for filter
What I want to accomplish is when my value is 1.1, I want for it to display 1.10.can i use, point num-round to some function that will round my number to 2 decimal pointsFor example<input  num-round='vm.round' />, where vm.round is a method on my controller that will round the number.
is that how you use it?
